extensions/hibernate-orm/deployment-spi/src/main/java/io/quarkus/hibernate/orm/deployment/spi/AdditionalJpaModelBuildItem.java

```diff
@@ -12,13 +13,35 @@
 public final class AdditionalJpaModelBuildItem extends MultiBuildItem {
 
     private final String className;
+    private final Set<String> persistenceUnits;
 
+    /**
+     * @deprecated Use {@link AdditionalJpaModelBuildItem#AdditionalJpaModelBuildItem(String, Set)} instead,
+     *             which should fit the use case of JBeret better.
+     */
+    @Deprecated(since = "3.26", forRemoval = true)
     public AdditionalJpaModelBuildItem(String className) {
+        this(className, null);
+    }
+
+    /**
+     * @param className The name of the additional class.
+     * @param persistenceUnits The name of persistence units to which this class should be added
+     *        even if the application does not request it explicitly (e.g. using `quarkus.hibernate-orm.packages`).
+     *        Note the class can still be added to a persistence unit at static init through other means --
+     *        for example Hibernate Envers and Hibernate Search use {@link org.hibernate.boot.spi.AdditionalMappingContributor}.
+     */
+    public AdditionalJpaModelBuildItem(String className, Set<String> persistenceUnits) {
         Objects.requireNonNull(className);
         this.className = className;
+        this.persistenceUnits = persistenceUnits;
```

Comment:
Minor nitpick, but if we're considering that `persistenceUnits` is only `null` when the legacy constructor is used, then we should use a `Objects.requireNonNull(persistenceUnits)` here and avoid calling `this` in the legacy constructor

Reply:
Good idea, thanks. Done!